segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainerTests.java

```diff
@@ -507,7 +509,7 @@ public static TestContext createContext(ScheduledExecutorService scheduledExecut
         TestContext(ScheduledExecutorService scheduledExecutorService) {
             this.storageFactory = new InMemoryStorageFactory(scheduledExecutorService);
             this.dataLogFactory = new InMemoryDurableDataLogFactory(MAX_DATA_LOG_APPEND_SIZE, scheduledExecutorService);
-            this.cacheStorage = new DirectMemoryCache(Integer.MAX_VALUE / 5);
```

Comment:
Precondition mismatch only happens during io.pravega.test.integration.RestoreBackUpDataRecoveryTest > testDurableDataLogFailRecoveryWatermarking . This test is different from other tests in the way that we write more data (x2) here. I tried increasing the cache storage from `Integer.MAX_VALUE / 5`  to `Integer.MAX_VALUE` , tried running several times, and the issue has not been reproduced yet.

Reply:
Is this change really required ?

Reply:
It helped alleviate `CacheStorage` getting closed issue (which eventually causes Preconditions mismatch).